pkg/operator/ceph/object/admin.go

```diff
@@ -65,10 +72,18 @@ func NewContext(context *clusterd.Context, clusterInfo *cephclient.ClusterInfo,
 }
 
 func NewMultisiteContext(context *clusterd.Context, clusterInfo *cephclient.ClusterInfo, store *cephv1.CephObjectStore) (*Context, error) {
-	objContext := &Context{Context: context, Name: store.Name, clusterInfo: clusterInfo}
-	realmName, zoneGroupName, zoneName, err := getMultisiteForObjectStore(context, store)
+	nsName := fmt.Sprintf("%s/%s", store.Namespace, store.Name)
+
+	objContext := NewContext(context, clusterInfo, store.Name)
+	objContext.UID = string(store.UID)
+
+	if err := UpdateContext(objContext, &store.Spec); err != nil {
+		return nil, err
+	}
```

Comment:
Making sure this function always fills in all possible fields of the returned context.